lib/utils/encryptor.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const crypto = require('crypto');
+const buffer1 = require('buffer');
+
+const Encryptor = {
+  /**
+   * Calculate a hash of the concatenated buffers with the given algorithm.
+   * @param {string} algorithm - The hash algorithm.
+   * @param {Array.<Buffer>} buffers - The buffers to concat and hash
+   * @returns {Buffer} The hash
+   */
+  hash(algorithm) {
+    const buffers = [];
+    for (let i = 1; i < arguments.length; i++) {
+      buffers[i - 1] = arguments[i];
+    }
+    algorithm = algorithm.toLowerCase();
```

Comment:
There's no need to execute this line (and the following 4 lines) every time we call hash(). This check should really move down to the convertPasswordToHash() function

Reply:
Can you make a suggestion. Could not get your point.

Reply:
The point is - this hash() function is called 100000 times by convertPasswordToHash(). The expression hashes.indexOf(algorithm) is running a linear array search every single time and is unnecessary if the check is performed within the convertPasswordToHash() function.

Reply:
Ok I see your point. Change has been implemented.